### PR TITLE
[15.0][MIG] base_sequence_option

### PR DESCRIPTION
**Pull request by Rad0van**
This would be a standard migration if only... I cannot force Odoo to run tests for this. Not in 15.0, not in 14.0 Anybody's got any idea why? So this is sort of an blind attempt based only on fact it seems to work together with _account_sequence_option_ that I plan to port next. Had to add return statement here:
```
    @classmethod
    def tearDownClass(cls):
        cls.loader.restore_registry()
        return super(CommonBaseSequenceOption, cls).tearDownClass()
```
That's pretty strange because such code exists in Odoo itself (didn't look long enough to find such thing in OCA). Also had to remove `string` attribute from `tree` element in view definition as per https://github.com/OCA/oca-addons-repo-template/issues/76 It was nagging as well.

Pylint was nagging me with 
```base_sequence_option/tests/common.py:91: [W8110(missing-return), CommonBaseSequenceOption.tearDownClass] Missing `return` (`super` is used) in method tearDownClass.```

Any feedback welcome.

### CODE REVIEW

**Comment by Rad0van** [comment]
@kittiu would you have a look here please? Alsowhat does that ci/runboat failing at the beginning mean? @gurneyalex ?

**Comment by kittiu** [comment]
I can not see any runbot log. Can you somehow try to push again. May be you can squash commits and try again.
I am not sure what it is but some time I experiences ci/runbot just behave like that.
But as other test are passing, I believe it is eligible for merge too, if reviewed.

**Comment by Rad0van** [comment]
> I can not see any runbot log. Can you somehow try to push again. May be you can squash commits and try again. I am not sure what it is but some time I experiences ci/runbot just behave like that. But as other test are passing, I believe it is eligible for merge too, if reviewed.

Even force-push didn't help. I cannot see anything there as well. Maybe @gurneyalex would know?

**Comment by Rad0van** [comment]
@kittiu can you see the tests being run? Because I cannot. The same happened to me at my local environments (14.0 and 15.o as well).

**Comment by gurneyalex** [comment]
runbot failing: please ignore, this is old story. Runboat seems to be running fine.

**Comment by gurneyalex** [comment]
To manually  test on runboat: 
* click on the "Details" link next to the runboat/build check
* Click on the start button on that page if you are not getting on an Odoo page directly, wait ~5s and then click on the "live" link -> this will get you to the Odoo login page.

**Comment by Saran440** [comment]
@Rad0van Hello, do you still do this pr?

**Comment by Rad0van** [comment]
> @Rad0van Hello, do you still do this pr?

Well what more can I do here?

**Comment by Saran440** [comment]
@Rad0van runbot is gone. may be you can up-to-date code and push it again.

**Comment by Rad0van** [comment]
> @Rad0van runbot is gone. may be you can up-to-date code and push it again.

Done ;-)

**Review by Saran440 — APPROVED** [review]
Functional Test with https://github.com/OCA/hr-expense/pull/149
LGTM :+1:

File: base_sequence_option/__manifest__.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    "website": "https://github.com/OCA/server-tools",
+    "category": "Tools",
+    "depends": ["base"],
+    "external_dependencies": {"python": ["odoo_test_helper"]},
+    "data": [
+        "security/ir.model.access.csv",
+        "security/sequence_option_security.xml",
+        "views/sequence_option_view.xml",
+    ],
+    "license": "LGPL-3",
```
Comment (Saran440) [inline base_sequence_option/__manifest__.py:20]:
```suggestion
    "license": "AGPL-3",
```

This module should AGPL-3 ?

File: base_sequence_option/__manifest__.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License LGPL-3.0 or later (https://www.gnu.org/licenses/lgpl.html).
```
Comment (Saran440) [inline base_sequence_option/__manifest__.py:2]:
```suggestion
# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).

```

All of license should agpl?

Comment (Rad0van) [reply base_sequence_option/__manifest__.py:2]: @Saran440 according to https://www.odoo-community.org/resources/faq AGPLv3, LGPLv3 or GPLv3 are possible. Also: I am not in a position to just "change" license of existing module I'm migrating. Only authors can do I believe.

Comment (Saran440) [reply base_sequence_option/__manifest__.py:2]: okay, i will change it next PR.
cc: @kittiu

File: base_sequence_option/__manifest__.py [hunk 3]
```diff
@@ -4,13 +4,14 @@
...
     "summary": "Alternative sequence options for specific models",
-    "version": "14.0.1.0.0",
+    "version": "15.0.1.0.0",
     "author": "Ecosoft, Odoo Community Association (OCA)",
     "maintainers": ["kittiu"],
     "development_status": "Alpha",
     "website": "https://github.com/OCA/server-tools",
     "category": "Tools",
     "depends": ["base"],
+    "external_dependencies": {"python": ["odoo_test_helper"]},
```
Comment (etobella) [inline base_sequence_option/__manifest__.py:14]: You don't need to add this here, just add it to test-requirements.txt (or create it if it does not exist :smile: )

Comment (Rad0van) [reply base_sequence_option/__manifest__.py:14]: OK, line removed. The `test-requirements.txt` already contains the `odoo_test_helper` so didn't do anything there. But it now causes pre-commit to fail. Not really sure I understand what's going on there.

**Comment by Saran440** [comment]
@Rad0van Can you update your code again, please?

**Comment by patrickt-oforce** [comment]
@pedrobaeza hi, we tested this PR locally, and the pre-commit fail is due to wrong flake8 source url , that pointing to gitlab instead of github, can you make an OCABot rebase?

**Comment by patrickt-oforce** [comment]
@Rad0van i have made a pr versus your branch to fix pre-commit fails

**Comment by patrickt-oforce** [comment]
@pedrobaeza thank you

**Comment by matteoopenf** [comment]
@kittiu  @etobella  @Saran440  [Rad0van](https://github.com/Rad0van) with my collegue fix pre-commit and superseed https://github.com/OCA/server-tools/pull/2749 can you have a look?

**Comment by matteoopenf** [comment]
@Rad0van  @patrickt-oforce  have done a PR from your branch and it's that https://github.com/Rad0van/server-tools/pull/2 if you merge that the pre-commit works fine

**Comment by Rad0van** [comment]
> @Rad0van @patrickt-oforce have done a PR from your branch and it's that [Rad0van#2](https://github.com/Rad0van/server-tools/pull/2) if you merge that the pre-commit works fine

But that removes line from _requirements.txt_ from whole repository. Not really sure that's the way to go...

**Comment by matteoopenf** [comment]
> > @Rad0van @patrickt-oforce have done a PR from your branch and it's that [Rad0van#2](https://github.com/Rad0van/server-tools/pull/2) if you merge that the pre-commit works fine
> 
> But that removes line from _requirements.txt_ from whole repository. Not really sure that's the way to go...

we have done a local test and works fine

**Comment by patrickt-oforce** [comment]
> > @Rad0van @patrickt-oforce have done a PR from your branch and it's that [Rad0van#2](https://github.com/Rad0van/server-tools/pull/2) if you merge that the pre-commit works fine
> 
> But that removes line from _requirements.txt_ from whole repository. Not really sure that's the way to go...

hi @Rad0van this deletion of requirements is operated by pre-commit, not our initiative; if you want can check results at #2749

**Comment by matteoopenf** [comment]
@Rad0van  the pre-commit now works fine now you should squash  the commit and we can ask the merge

**Comment by matteoopenf** [comment]
@etobella  the pre-commit is now fix

**Comment by Rad0van** [comment]
> @kittiu @etobella @Saran440 [Rad0van](https://github.com/Rad0van) with my collegue fix pre-commit and superseed #2749 can you have a look?

Isn't it rather the ocabot rebase fixed the precommit?

**Review by matteoopenf — APPROVED** [review]

**Comment by Rad0van** [comment]
Sorry for the commit/squash mess. Have tried removing that `"external_dependencies": {"python": ["odoo_test_helper"]}` line from manifest but it keeps breaking despite that library being part of test-requirements.txt: https://github.com/OCA/server-tools/blob/6d2ffc7d289b99a54ceb11352f7fc10839e40aee/test-requirements.txt#L1

**Comment by matteoopenf** [comment]
> Sorry for the commit/squash mess. Have tried removing that `"external_dependencies": {"python": ["odoo_test_helper"]}` line from manifest but it keeps breaking despite that library being part of test-requirements.txt:
> 
> https://github.com/OCA/server-tools/blob/6d2ffc7d289b99a54ceb11352f7fc10839e40aee/test-requirements.txt#L1

have no sense is done by pre-commit.
So, anyway after that is ready to merge @Rad0van  what do you think?

**Comment by Rad0van** [comment]
> > Sorry for the commit/squash mess. Have tried removing that `"external_dependencies": {"python": ["odoo_test_helper"]}` line from manifest but it keeps breaking despite that library being part of test-requirements.txt:
> > https://github.com/OCA/server-tools/blob/6d2ffc7d289b99a54ceb11352f7fc10839e40aee/test-requirements.txt#L1
> 
> have no sense is done by pre-commit. So, anyway after that is ready to merge @Rad0van what do you think?

Now it passes. I have squashed all the commits so proper order is maintained. I think the rule is 2 approvals and 5 (or 7) days. Someone with rights need to merge it. I do not have those rights...

**Review by patrickt-oforce — APPROVED** [review]

**Comment by matteoopenf** [comment]
@pedrobaeza I think now is ready to merge.
Do you agree, or need something this pr?

**Comment by patrickt-oforce** [comment]
@Saran440 i think your PR #2743 and mine can be close; this PR now seem fixed and all check has green mark. About this PR [OCA/account-financial-tools#1635](/OCA/account-financial-tools/pull/1635) can you try to push an empty commit to restart CI?

**Comment by pedrobaeza** [comment]
Merging according reviews.

/ocabot migration base_sequence_option
/ocabot merge nobump

**Pull request merged**